File: HackerRankAlgorithmsTrack/compareTheTriples.py
#!/bin/python3

import os
import sys

def compare(a, b):
    if a > b : return (1,0)
    elif a < b : return (0, 1)
    else : return (0,0)
# Complete the solve function below.
#
def solve(a0, a1, a2, b0, b1, b2):
    return map(sum, zip(*[compare(a0, b0), compare(a1, b1), compare(a2, b2)]))

    
if __name__ == '__main__':
    f = open(os.environ['OUTPUT_PATH'], 'w')

    a0A1A2 = input().split()

    a0 = int(a0A1A2[0])

    a1 = int(a0A1A2[1])

    a2 = int(a0A1A2[2])

    b0B1B2 = input().split()

    b0 = int(b0B1B2[0])

    b1 = int(b0B1B2[1])

    b2 = int(b0B1B2[2])

    result = solve(a0, a1, a2, b0, b1, b2)

    f.write(' '.join(map(str, result)))
    f.write('\n')

    f.close()

File: hackerRank/algorithms/compareTheTriples.hs
{-# LANGUAGE FlexibleInstances, UndecidableInstances, DuplicateRecordFields #-}

module Main where

import Control.Monad
import Data.List
import Data.List.Split
import Data.Set
import System.Environment
import System.IO

--
-- Complete the solve function below.
--
solve a0 a1 a2 b0 b1 b2 = do
  print a0
    --
    -- Write your code here.
    --

main :: IO()
main = do
    outputPath <- getEnv "OUTPUT_PATH"
    fptr <- openFile outputPath WriteMode

    a0A1A2Temp <- getLine
    print a0A1A2Temp
    let a0A1A2 = words a0A1A2Temp

    let a0 = read (a0A1A2 !! 0) :: Int

    let a1 = read (a0A1A2 !! 1) :: Int

    let a2 = read (a0A1A2 !! 2) :: Int

    b0B1B2Temp <- getLine
    let b0B1B2 = words b0B1B2Temp

    let b0 = read (b0B1B2 !! 0) :: Int

    let b1 = read (b0B1B2 !! 1) :: Int

    let b2 = read (b0B1B2 !! 2) :: Int

    {-let result = solve a0 a1 a2 b0 b1 b2-}

    {-hPutStrLn fptr $ intercalate " " $ Data.List.map (\x -> show x) $ result-}

    hFlush fptr
    hClose fptr
